src/value.rs

```diff
@@ -198,15 +217,34 @@ impl ValueSum {
             sign,
         )
     }
+
+    pub(crate) fn into<V: TryFrom<i64>>(self) -> Result<V, BuildError> {
```

Comment:
str4d ([Link](https://github.com/zcash/orchard/pull/471#discussion_r2542756837)): Rename this; we should never have a manual `into()` method that aliases the `Into` trait while using a different method signature (which is closer to `TryInto::try_into`).

```suggestion
    pub(crate) fn into_value_balance<V: TryFrom<i64>>(self) -> Result<V, BuildError> {
```